api/core/app/apps/advanced_chat/app_generator.py

```diff
@@ -276,15 +278,24 @@ def _generate_worker(self, flask_app: Flask,
                     # get conversation and message
                     conversation = self._get_conversation(conversation_id)
                     message = self._get_message(message_id)
+                    app_record = db.session.query(App).filter(App.id == application_generate_entity.app_config.app_id).first()
+                    if not app_record:
+                        raise ValueError("App not found")
 
                     # chatbot app
-                    runner = AdvancedChatAppRunner()
-                    runner.run(
-                        application_generate_entity=application_generate_entity,
-                        queue_manager=queue_manager,
-                        conversation=conversation,
-                        message=message
-                    )
+                    try:
+                        max_active_request = self._get_max_active_requests(app_record)
```

Comment:
The rate limit logic would be more appropriately placed in the `generate()` func above, where it can be intercepted directly without having to think about the `Iterator` return.

Reply:
As well as avoiding duplicate queries for app records, and dealing with stream output below